qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -53,12 +53,31 @@ def _choose_euler_basis(basis_gates):
     basis_set = set(basis_gates or [])
 
     for basis, gates in one_qubit_decompose.ONE_QUBIT_EULER_BASIS_GATES.items():
+
         if set(gates).issubset(basis_set):
             return basis
 
     return None
 
 
+def _choose_bases(basis_gates, basis_dict=None):
+    """Find the matching basis string keys from the list of basis gates from the backend."""
+    if basis_gates is None:
+        basis_set = set()
+    else:
+        basis_set = set(basis_gates)
+
+    if basis_dict is None:
+        basis_dict = one_qubit_decompose.ONE_QUBIT_EULER_BASIS_GATES
+
+    out_basis = []
+    for basis, gates in basis_dict.items():
+        if set(gates).issubset(basis_set):
+            return out_basis.append(basis)
+
+    return out_basis
```

Comment:
This function can return the empty list.  We probably should define what's expected of a plugin if that happens (just returns `None` immediately?), or is it better to detect it and decay to using the default handler in those cases without ever calling the plugin?  Either way, probably needs documenting in the `supported_bases` docstring.

Reply:
I'm just going to document this as returning an empty list in this case. The logic for handling it feels like it should be up to the particular plugin. The default plugin handles this case by doing nothing.

Reply:
Yep, that sounds entirely reasonable to me.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/6124/commits/9ca591a6bc6cd32101775edd0f20bd4546770c64